src/test/cypress/e2e/exam/test-exam/TestExamTestRun.cy.ts

```diff
@@ -0,0 +1,154 @@
+import { Exam } from 'app/entities/exam.model';
+import { CypressExamBuilder, convertCourseAfterMultiPart } from '../../../support/requests/CourseManagementRequests';
+import dayjs from 'dayjs/esm';
+import submission from '../../../fixtures/exercise/programming/build_error/submission.json';
+import { Course } from 'app/entities/course.model';
+import { generateUUID } from '../../../support/utils';
+import { EXERCISE_TYPE } from '../../../support/constants';
+import { Exercise } from 'src/test/cypress/support/pageobjects/exam/ExamParticipation';
+import { Interception } from 'cypress/types/net-stubbing';
+import { courseManagementRequest, examExerciseGroupCreation, examManagement, examNavigation, examParticipation, examTestRun } from '../../../support/artemis';
+import { admin, instructor } from '../../../support/users';
+
+// Common primitives
+const textFixture = 'loremIpsum.txt';
+const examTitle = 'exam' + generateUUID();
+
+const exerciseArray: Array<Exercise> = [];
+
+describe('Test exam test run', () => {
+    let course: Course;
+    let exam: Exam;
+    let testRun: any;
+
+    before(() => {
+        cy.login(admin);
+        courseManagementRequest.createCourse(true).then((response) => {
+            course = convertCourseAfterMultiPart(response);
+            const examContent = new CypressExamBuilder(course)
+                .title(examTitle)
+                .testExam()
+                .visibleDate(dayjs().subtract(3, 'days'))
+                .startDate(dayjs().add(1, 'days'))
+                .endDate(dayjs().add(3, 'days'))
+                .examMaxPoints(40)
+                .numberOfExercises(4)
+                .build();
+            courseManagementRequest.createExam(examContent).then((examResponse) => {
+                exam = examResponse.body;
+                examExerciseGroupCreation.addGroupWithExercise(exerciseArray, exam, EXERCISE_TYPE.Text, { textFixture });
+
+                examExerciseGroupCreation.addGroupWithExercise(exerciseArray, exam, EXERCISE_TYPE.Programming, { submission, practiceMode: true });
+
+                examExerciseGroupCreation.addGroupWithExercise(exerciseArray, exam, EXERCISE_TYPE.Quiz, { quizExerciseID: 0 });
+
+                examExerciseGroupCreation.addGroupWithExercise(exerciseArray, exam, EXERCISE_TYPE.Modeling);
+            });
+        });
+    });
+
+    beforeEach('Create test run instance', () => {
+        cy.login(instructor);
+        // API call does not work yet, for now we use the UI to create the test run
```

Comment:
(optional) Is this actually a TODO for the future? :D
```suggestion
        // TODO: API call does not work yet, for now we use the UI to create the test run
```